skimage/feature/_texture.pyx

```diff
@@ -264,3 +265,89 @@ def _local_binary_pattern(double[:, ::1] image,
                 output[r, c] = lbp
 
     return np.asarray(output)
+
+
+# Constant values that are used by `multiblock_local_binary_pattern` function.
+# Values represent offsets of neighbour rectangles relative to central one.
+# It has order starting from top left and going clockwise.
+cdef:
+    Py_ssize_t[::1] mlbp_x_offsets = np.asarray([-1, 0, 1, 1, 1, 0, -1, -1])
+    Py_ssize_t[::1] mlbp_y_offsets = np.asarray([-1, -1, -1, 0, 1, 1, 1, 0])
+
+def _multiblock_lbp(float[:, ::1] int_image,
+                    Py_ssize_t x,
+                    Py_ssize_t y,
+                    Py_ssize_t width,
+                    Py_ssize_t height):
+    """Multi-block local binary pattern.
+
+    Parameters
+    ----------
+    int_image : (N, M) float array
+        Integral image.
+    x : int
+        X-coordinate of top left corner of a rectangle containing feature.
+    y : int
+        Y-coordinate of top left corner of a rectangle containing feature.
+    width : int
+        Width of one of 9 equal rectangles that will be used to compute
+        a feature.
+    height : int
+        Height of one of 9 equal rectangles that will be used to compute
+        a feature.
+
+    Returns
+    -------
+    output : int
+        8-bit MB-LBP feature descriptor.
+
+    References
+    ----------
+    .. [1] Face Detection Based on Multi-Block LBP
+           Representation. Lun Zhang, Rufeng Chu, Shiming Xiang, Shengcai Liao,
+           Stan Z. Li
+           http://www.cbsr.ia.ac.cn/users/scliao/papers/Zhang-ICB07-MBLBP.pdf
+    """
+
+    # Top-left coordinates of central rectangle
+    cdef:
+        Py_ssize_t central_rect_x = x + width
+        Py_ssize_t central_rect_y = y + height
+
+    # Sum of intensity values of central rectangle
+    cdef float central_rect_val = integrate(int_image, central_rect_y, central_rect_x,
+                                            central_rect_y + height - 1,
+                                            central_rect_x + width - 1)
+
+    cdef:
+        Py_ssize_t element_num, offset_x, offset_y
+        Py_ssize_t current_rect_x, current_rect_y
+        double current_rect_val
+        int has_greater_value
+        int lbp_code = 0
+
+    for element_num in range(8):
+
+        offset_x = mlbp_x_offsets[element_num]
+        offset_y = mlbp_y_offsets[element_num]
+
+
+        current_rect_x = central_rect_x + offset_x * width
```

Comment:
@JDWarner yes it will improve the performance. What I am about to do to correct it:
create new array, run small loop to populate it with values from `mlbp_x_offsets` that are multiplied with `width`. Or I can actually allocate two dummy arrays for each premultiplied set of offsets and just populate them all over again on each function call. Do you think that it's worth it only for 8 loop iterations? And if yes which approach of described two you prefer more? Thank you